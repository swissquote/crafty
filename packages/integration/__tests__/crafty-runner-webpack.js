/* global jest, describe, it, expect */

const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

it("Compiles Only with Webpack", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-runner-webpack/compiles"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/690.myBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/690.myBundle.min.js.map")).toBeTruthy();

  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myBundle.min.js")
  ).toMatchSnapshot();
  expect(
    testUtils.readForSnapshot(cwd, "dist/js/690.myBundle.min.js")
  ).toMatchSnapshot();
});

it("Fails gracefully on broken markup", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-runner-webpack/fails"
  );

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});
