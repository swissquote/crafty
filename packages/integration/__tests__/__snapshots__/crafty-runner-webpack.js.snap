// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles Only with Webpack 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:

[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
asset myBundle.min.js ___ KiB [emitted] [minimized] (name: default)
  sourceMap myBundle.min.js.map ___ KiB [emitted] [dev] (auxiliary name: default)
asset 690.myBundle.min.js ___ bytes [emitted] [minimized]
  sourceMap 690.myBundle.min.js.map ___ bytes [emitted] [dev]
Entrypoint default ___ KiB (___ KiB) = myBundle.min.js 1 auxiliary asset
chunk (runtime: default) 690.myBundle.min.js ___ bytes [rendered]
  ./js/SomeClass.js ___ bytes [built] [code generated]
chunk (runtime: default) myBundle.min.js (default) ___ bytes (javascript) ___ KiB (runtime) [entry] [rendered]
  runtime modules ___ KiB 9 modules
  ./js/script.js ___ bytes [built] [code generated]

  Compiled successfully!
  Δt ____ms

[__:__:__] Finished 'js_myBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Compiles Only with Webpack 2`] = `
"(function(e,t){if(\\"object\\"===typeof exports&&\\"object\\"===typeof module)module.exports=t();else if(\\"function\\"===typeof define&&define.amd)define([],t);else{var r=t();for(var n in r)(\\"object\\"===typeof exports?exports:e)[n]=r[n]}})(self,(function(){return function(){\\"use strict\\";var e={450:function(e,t,r){r.r(t),r.d(t,{default:function(){return n}});class n{constructor(){console.log(\\"init class\\")}method(){r.e(690).then(r.bind(r,690)).then((function(e){const t=new e.default(\\"Bill\\");t.sayHi()}))}}}};var t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={exports:{}};return e[n](o,o.exports,r),o.exports}return r.m=e,r.d=function(e,t){for(var n in t)r.o(t,n)&&!r.o(e,n)&&Object.defineProperty(e,n,{enumerable:!0,get:t[n]})},r.f={},r.e=function(e){return Promise.all(Object.keys(r.f).reduce((function(t,n){return r.f[n](e,t),t}),[]))},r.u=function(e){return e+\\".myBundle.min.js\\"},r.g=function(){if(\\"object\\"===typeof globalThis)return globalThis;try{return this||new Function(\\"return this\\")()}catch(e){if(\\"object\\"===typeof window)return window}}(),r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},function(){var e={};var t=\\"UNIQID:\\";r.l=function(n,o,i){if(e[n])return void e[n].push(o);var u,a;if(void 0!==i){var c=document.getElementsByTagName(\\"script\\");for(var f=0;f<c.length;f++){var l=c[f];if(l.getAttribute(\\"src\\")==n||l.getAttribute(\\"data-webpack\\")==t+i){u=l;break}}}u||(a=!0,u=document.createElement(\\"script\\"),u.charset=\\"utf-8\\",u.timeout=120,r.nc&&u.setAttribute(\\"nonce\\",r.nc),u.setAttribute(\\"data-webpack\\",t+i),u.src=n),e[n]=[o];var s=function(t,r){u.onerror=u.onload=null,clearTimeout(p);var o=e[n];if(delete e[n],u.parentNode&&u.parentNode.removeChild(u),o&&o.forEach((function(e){return e(r)})),t)return t(r)};var p=setTimeout(s.bind(null,void 0,{type:\\"timeout\\",target:u}),12e4);u.onerror=s.bind(null,u.onerror),u.onload=s.bind(null,u.onload),a&&document.head.appendChild(u)}}(),r.r=function(e){\\"undefined\\"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},function(){var e;r.g.importScripts&&(e=r.g.location+\\"\\");var t=r.g.document;if(!e&&t&&(t.currentScript&&(e=t.currentScript.src),!e)){var n=t.getElementsByTagName(\\"script\\");n.length&&(e=n[n.length-1].src)}if(!e)throw new Error(\\"Automatic publicPath is not supported in this browser\\");e=e.replace(/#.*$/,\\"\\").replace(/\\\\?.*$/,\\"\\").replace(/\\\\/[^\\\\/]+$/,\\"/\\"),r.p=e}(),function(){var e={913:0};r.f.j=function(t,n){var o=r.o(e,t)?e[t]:void 0;if(0!==o)if(o)n.push(o[2]);else{var i=new Promise((function(r,n){o=e[t]=[r,n]}));n.push(o[2]=i);var u=r.p+r.u(t);var a=new Error;var c=function(n){if(r.o(e,t)&&(o=e[t],0!==o&&(e[t]=void 0),o)){var i=n&&(\\"load\\"===n.type?\\"missing\\":n.type);var u=n&&n.target&&n.target.src;a.message=\\"Loading chunk \\"+t+\\" failed.\\\\n(\\"+i+\\": \\"+u+\\")\\",a.name=\\"ChunkLoadError\\",a.type=i,a.request=u,o[1](a)}};r.l(u,c,\\"chunk-\\"+t)}};var t=function(t){var n=t[0];var i=t[1];var u=t[2];var a,c,f=0,l=[];for(;f<n.length;f++)c=n[f],r.o(e,c)&&e[c]&&l.push(e[c][0]),e[c]=0;for(a in i)r.o(i,a)&&(r.m[a]=i[a]);u&&u(r),o(t);while(l.length)l.shift()()};var n=self.webpackChunkUNIQID=self.webpackChunkUNIQID||[];var o=n.push.bind(n);n.push=t}(),r(450)}()}));
//# sourceMappingURL=myBundle.min.js.map"
`;

exports[`Compiles Only with Webpack 3`] = `
"(self.webpackChunkUNIQID=self.webpackChunkUNIQID||[]).push([[690],{690:function(s,e,t){\\"use strict\\";t.r(e),t.d(e,{default:function(){return c}});class c{constructor(s){this.field=s}sayHi(){console.log(this.field)}}}}]);
//# sourceMappingURL=690.myBundle.min.js.map"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:

[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...
assets by status ___ bytes [cached] 1 asset
Entrypoint default = myBundle.min.js 1 auxiliary asset
chunk (runtime: default) myBundle.min.js (default) ___ bytes [entry] [rendered]
  ./js/script.js ___ bytes [built] [code generated] [1 error]

  Failed to compile.

Module parse failed: Unexpected token (15:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|
| }
>

  Δt ____ms ✖ 1 problem (1 error, 0 warnings)

[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Webpack compilation failed
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;
